src/core.js

```diff
@@ -425,6 +425,16 @@ jQuery.extend({
 	support: support
 });
 
+// jsHint would error on this code due to the Symbol not being defined in ES5.
```

Comment:
@rwaldron 
>> jsHint would error on this code due to the Symbol not being defined in ES5.
>
> How problematic is it for jQuery to switch on `esnext: true`?

This would be going even further; we'd get automatically defined all ES6 globals & syntax which not all our supported browsers implement.

On the `compat` branch we even explicitly define `es3: true` so that jsHint won't e.g. allow us to introduce trailing commas in array/object literals (yes, it did happen).

I think turning `esnext` on could be dangerous here.